src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -54,40 +56,38 @@ public TableStorageScaleMetricsRepository(IConfiguration configuration, IHostIdP
             _delegatingHandlerProvider = delegatingHandlerProvider ?? throw new ArgumentNullException(nameof(delegatingHandlerProvider));
         }
 
-        internal CloudTableClient TableClient
+        internal TableServiceClient TableClient
         {
             get
             {
-                if (_tableClient == null)
+                if (_tableClient is null)
                 {
                     string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
-                    if (!string.IsNullOrEmpty(storageConnectionString) &&
-                        CloudStorageAccount.TryParse(storageConnectionString, out CloudStorageAccount account))
-                    {
-                        var restConfig = new RestExecutorConfiguration { DelegatingHandler = _delegatingHandlerProvider.Create() };
-                        var tableClientConfig = new TableClientConfiguration { RestExecutorConfiguration = restConfig };
```

Comment:
You're removing the delegating handler and other logic (potentially leaving DefaultDelegatingHandlerProvider an unused type?). @brettsam added this logic - wondering what issues may arise if we remove this?

Reply:
This is still pending, I initially removed it to test everything else. I wanted to discuss what's the best way of incorporating the DelegatingHandler into the new SDK. By default it will use a shared HttpClientTransport (same HttpClient) for all the clients.

From my initial investigation we would need to provide an HttpClient that uses our DelegatingHandler when we instantiate the TableServiceClient via the TableClientOptions. 

I also think that we should check the way we instantiate these clients, same as for Diagnostic Events, specially if we need to enable Managed Identity support instead of using the connection string for these scenarios.

https://github.com/Azure/azure-functions-host/blob/49c49c9ea2153515a2d7a2de71a43f4dd80cb856/src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs#L68

Reply:
I ended up simplifying a bit the `DelegatingHandler` and passing it to the `TableServiceClient `via `ClientOptions` (just for consumption), at least for now. 

I think we should better discuss adding a new `TableStorageProvider` and replace the manual creation of the clients (here and for DiagnosticEvents) in a follow-up PR, for example things such: do we need to add a `IAzureTableStorageProvider` in the [Webjobs.Host.Storage package](https://github.com/Azure/azure-webjobs-sdk/tree/dev/src/Microsoft.Azure.WebJobs.Host.Storage)?

@fabiocav

Reply:
@cjaliaga using tables is an implementation detail here, we wouldn't want to expose a public contract just for providing the clients.

Instead, there is already a public contract for this: `IAzureClientFactory<TClient>` from `Microsoft.Extensions.Azure`. We can evaluate using this to extract client construction to DI in a later PR.

Looking at the existing delegating handler, the comment is that it is "because connections are limited". I think what they are referring to in connection pooling: re-use existing HttpClients instead of instantiating new ones repeatedly. As @cjaliaga mentioned, the new track2 clients have made great improvements here. I do wonder if it is even needed anymore?